py/desispec/io/meta.py

```diff
@@ -178,6 +190,18 @@ def findfile(filetype, night=None, expid=None, camera=None,
         exists; otherwise it returns the normal read/write path.
     """
     log = get_logger()
+
```

Comment:
The behavior of
> The code will now raise an IOError if the alternative compressed/uncompressed file exists unless the option
> readonly=True is set in checkgzip or findfile routines.

is new and potentially surprising, so please add a note about that in the `findfile` docstring, clarifying the relationship and consequences between the readonly option, $DESI_COMPRESSION, and the possible pre-existence of a file.  I think the chain of logic is
```
filepath = canonical location considering $DESI_COMPRESSION
if alternate (non)compressed file exists:
    if readonly:
        ok, use alternate path for filepath
    else:
        raise IOError  # due to pre-existing non-canonical file

return filepath
```

Is that correct and/or can we express it more simply?

Reply:
Yes it's correct. I have added a sentence in the Notes paragraph of findfile docstring,